src/Microsoft.ML.Data/Prediction/Calibrator.cs

```diff
@@ -116,59 +116,82 @@ internal interface ISelfCalibratingPredictor
         IPredictor Calibrate(IChannel ch, IDataView data, ICalibratorTrainer caliTrainer, int maxRows);
     }
 
-    public abstract class CalibratedPredictorBase :
+    /// <summary>
+    /// <see cref="IWeaklyTypedCalibratedPredictor"/> provides a weekly-typed way to access strongly-typed
+    /// <see cref="CalibratedPredictorBase{TSubPredictor, TCalibrator}.SubModelParameters"/> and
+    /// <see cref="CalibratedPredictorBase{TSubPredictor, TCalibrator}.Calibrator"/>.
+    /// <see cref="IWeaklyTypedCalibratedPredictor"/> is commonly used in weekly-typed expressions. The
+    /// existence of this interface is just for supporting existing codebase, so we discourage its uses.
+    /// </summary>
+    [BestFriend]
+    internal interface IWeaklyTypedCalibratedPredictor
+    {
+        IPredictorProducing<float> WeeklyTypedSubModelParameters { get; }
+        ICalibrator WeeklyTypedCalibrator { get; }
+    }
+
+    public abstract class CalibratedPredictorBase<TSubModelParameters, TCalibrator> :
```

Comment:
>CalibratedPredictorBase [](start = 26, length = 23)

@tfinley@gmail.com since we transitioned to a place where Predictors are named ModelParameters, should we rename this class `CalibratedModelParametersBase`?

Reply:
Sounds good. I am changing it.

---
In reply to: [255160688](https://github.com/dotnet/machinelearning/pull/2435#discussion_r255160688) [](ancestors = 255160688)

Reply:
Potentially the same should apply to the derived classes.

---
In reply to: [255160688](https://github.com/dotnet/machinelearning/pull/2435#discussion_r255160688) [](ancestors = 255160688)